ONX.CRM.BLL/Services/GroupService.cs

```diff
@@ -12,5 +16,60 @@ public GroupService(IRepository<Group> repository) : base(repository)
         {
             
         }
+
+        public async Task<IEnumerable<Group>> SearchGroups(string query, string status)
+        {
+            if (!string.IsNullOrEmpty(query))
+            {
+                return await GetGroupsByQuery(query);
+            }
+
+            if (!string.IsNullOrEmpty(status))
+            {
+                return await GetGroupsByStatus(status);
+            }
+
+            return null;
+        }
+
+        private async Task<IEnumerable<Group>> GetGroupsByStatus(string status)
+        {
+            var groupsList = (await _repository.GetAllAsync()).Where(s => s.Status.ToString() == status);
+            return groupsList;
+        }
+
+        private async Task<IEnumerable<Group>> GetGroupsByQuery(string query)
+        {
+            var groupsList = await _repository.GetAllAsync();
```

Comment:
Should be used IQueryable Where

Reply:
I figured out that IQueryable we use this in the repository. and in the services we use IEnumerable. Because changing to IQueryable we need to change and in action.